### ISSUE TITLE
想请教一下训练过程运行到restore model后卡住不显示的问题

### ISSUE THREAD

**Issue by marshmallow1412** [Ixiaohuihuihui/Tiny-Defect-Detection-for-PCB#14]
您好，尝试用您项目中的数据集训练程序，将下载的部分数据集放入名为pcb的文件夹放入$PATH_ROOT/data中，xml文件放入pcb/Annotation,图片放入pcb/JPEGImage。运行divide_data.py和convert_data_to_tfrecord.py 都没问题。但运行train.py，打印出restore model后卡住，不显示报错也没使用gpu运行。单步调试发现进入restore model后面循环中的else语句_, global_stepnp, summary_str = sess.run([train_op, global_step, summary_op])卡住，不继续运行下一步。请问大佬可以帮忙看一下是什么原因？  谢谢！

`python train.py
++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--++--
/home/lkc/Project/Tiny_Defect_Detection_for_PCB_master
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tfplot/figure.py:18: The name tf.Summary is deprecated. Please use tf.compat.v1.Summary instead.
WARNING:tensorflow:From ../libs/networks/mobilenet/mobilenet.py:356: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.
tfrecord path is --> /home/lkc/Project/Tiny_Defect_Detection_for_PCB_master/data/tfrecord/pcb_train*
WARNING:tensorflow:From ../data/io/read_tfrecord.py:86: The name tf.train.match_filenames_once is deprecated. Please use tf.io.match_filenames_once instead.
WARNING:tensorflow:From ../data/io/read_tfrecord.py:88: string_input_producer (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensor_slices(string_tensor).shuffle(tf.shape(input_tensor, out_type=tf.int64)[0]).repeat(num_epochs)`. If `shuffle=False`, omit the `.shuffle(...)`.
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/training/input.py:278: input_producer (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensor_slices(input_tensor).shuffle(tf.shape(input_tensor, out_type=tf.int64)[0]).repeat(num_epochs)`. If `shuffle=False`, omit the `.shuffle(...)`.
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/training/input.py:190: limit_epochs (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensors(tensor).repeat(num_epochs)`.
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/training/input.py:199: QueueRunner.__init__ (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/training/input.py:199: add_queue_runner (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:From ../data/io/read_tfrecord.py:18: TFRecordReader.__init__ (from tensorflow.python.ops.io_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.TFRecordDataset`.
WARNING:tensorflow:From ../data/io/read_tfrecord.py:21: The name tf.parse_single_example is deprecated. Please use tf.io.parse_single_example instead.

WARNING:tensorflow:From ../data/io/read_tfrecord.py:24: The name tf.FixedLenFeature is deprecated. Please use tf.io.FixedLenFeature instead.

WARNING:tensorflow:From ../data/io/image_preprocess.py:33: The name tf.image.resize_bilinear is deprecated. Please use tf.compat.v1.image.resize_bilinear instead.

WARNING:tensorflow:From ../data/io/image_preprocess.py:73: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.

WARNING:tensorflow:From ../data/io/read_tfrecord.py:98: batch (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.batch(batch_size)` (or `padded_batch(...)` if `dynamic_pad=True`).
WARNING:tensorflow:From train.py:37: The name tf.no_regularizer is deprecated. Please use tf.compat.v1.no_regularizer instead.

WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tfplot/ops.py:114: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, there are two
    options available in V2.
    - tf.py_function takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means `tf.py_function`s can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
    - tf.numpy_function maintains the semantics of the deprecated tf.py_func
    (it is not differentiable, and manipulates numpy arrays). It drops the
    stateful argument making all functions stateful.

we are in Pyramid::-======>>>>
['P2', 'P3', 'P4', 'P5', 'P6']
base_anchor_size are:  [15, 25, 40, 60, 80]
________________________________________
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:446: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:452: The name tf.summary.image is deprecated. Please use tf.compat.v1.summary.image instead.

WARNING:tensorflow:From ../libs/networks/build_whole_network.py:463: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:473: to_int64 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:473: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:474: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.

WARNING:tensorflow:From ../libs/networks/build_whole_network.py:137: calling crop_and_resize_v1 (from tensorflow.python.ops.image_ops_impl) with box_ind is deprecated and will be removed in a future version.
Instructions for updating:
box_ind is deprecated, use box_indices instead
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/contrib/layers/python/layers/layers.py:1634: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From train.py:51: get_regularization_losses (from tensorflow.contrib.losses.python.losses.loss_ops) is deprecated and will be removed after 2016-12-30.
Instructions for updating:
Use tf.losses.get_regularization_losses instead.
WARNING:tensorflow:From train.py:90: get_or_create_global_step (from tensorflow.contrib.framework.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.get_or_create_global_step
WARNING:tensorflow:From train.py:91: The name tf.train.piecewise_constant is deprecated. Please use tf.compat.v1.train.piecewise_constant instead.

WARNING:tensorflow:From train.py:95: The name tf.train.MomentumOptimizer is deprecated. Please use tf.compat.v1.train.MomentumOptimizer instead.

/home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/ops/gradients_util.py:93: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
WARNING:tensorflow:From train.py:117: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.

Tensor("Merge/MergeSummary:0", shape=(), dtype=string)
WARNING:tensorflow:From ../libs/networks/build_whole_network.py:541: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.

model restore from : /home/lkc/Project/Tiny_Defect_Detection_for_PCB_master/output/trained_weights/FPN_Res101_0117_OHEM/pcb_30000model.ckpt
2020-02-25 11:56:54.021228: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-02-25 11:56:54.040171: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 4051785000 Hz
2020-02-25 11:56:54.040694: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5645ca937790 executing computations on platform Host. Devices:
2020-02-25 11:56:54.040706: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2020-02-25 11:56:54.042346: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
2020-02-25 11:56:54.189673: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.190299: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5645ca655490 executing computations on platform CUDA. Devices:
2020-02-25 11:56:54.190310: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): GeForce RTX 2080, Compute Capability 7.5
2020-02-25 11:56:54.190448: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.190993: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties:
name: GeForce RTX 2080 major: 7 minor: 5 memoryClockRate(GHz): 1.77
pciBusID: 0000:1c:00.0
2020-02-25 11:56:54.191181: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2020-02-25 11:56:54.191932: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
2020-02-25 11:56:54.192663: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
2020-02-25 11:56:54.192820: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
2020-02-25 11:56:54.193721: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
2020-02-25 11:56:54.194389: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
2020-02-25 11:56:54.196402: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2020-02-25 11:56:54.196511: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.197108: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.197633: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
2020-02-25 11:56:54.197669: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2020-02-25 11:56:54.198358: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-02-25 11:56:54.198368: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0
2020-02-25 11:56:54.198372: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N
2020-02-25 11:56:54.198547: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.199106: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-02-25 11:56:54.199650: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7044 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080, pci bus id: 0000:1c:00.0, compute capability: 7.5)
WARNING:tensorflow:From /home/lkc/.virtualenvs/tensor_gpu/lib/python3.7/site-packages/tensorflow/python/training/saver.py:1276: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
restore model
WARNING:tensorflow:From train.py:138: start_queue_runners (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:From train.py:142: The name tf.summary.FileWriter is deprecated. Please use tf.compat.v1.summary.FileWriter instead.

2020-02-25 11:56:57.604949: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.
2020-02-25 11:57:01.615422: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0`
